### PR TITLE
Feature: optional oauth2 integration

### PR DESCRIPTION
**Pull request by rzykie**
## Ticket - [12](https://github.com/orgs/mugna-tech/projects/30/views/1?pane=issue&itemId=81831553&issue=mugna-tech%7Creact-blueprint%7C12)

_Related tickets:_  
_Related PRs:_  

## Type of PR

- [ ] New Feature
- [ ] Bug Fix
- [ ] Minor Change

## Screenshots
oauth register path:
image

## Changes introduced

_Add django-oauth-toolkit_
_Update poetry lock file_
_Update settings to include oauth2 setup_
_Add urls for oauth2 provider_
_Add cookiecutter option for oauth2 integration_
_Add cleanup in post gen project_

### CODE REVIEW

File: {{cookiecutter.project_slug_dashed}}/{{cookiecutter.project_slug_snaked}}/urls.py [hunk 1]
```diff
@@ -17,6 +20,9 @@
 
 urlpatterns = [
     path("admin/", admin.site.urls),
+{%- if cookiecutter.use_oauth2_provider == "y" %}
+    path("o/", include(oauth2_urls, namespace="oauth2_provider")),
```
Comment (jlorencelim) [inline {{cookiecutter.project_slug_dashed}}/{{cookiecutter.project_slug_snaked}}/urls.py:24]: I think this should be included inside the api/v1. Same level as the auth

Comment (jacelo) [reply {{cookiecutter.project_slug_dashed}}/{{cookiecutter.project_slug_snaked}}/urls.py:24]: Also change `o` to `oauth2` to be more descriptive.

**Review by jacelo — APPROVED** [review]
Apply minor comments. Otherwise LGTM.

File: {{cookiecutter.project_slug_dashed}}/{{cookiecutter.project_slug_snaked}}/urls.py (hunk 1 above)
Comment (rzykie) [reply {{cookiecutter.project_slug_dashed}}/{{cookiecutter.project_slug_snaked}}/urls.py]: noted on this

**Review by jlorencelim — APPROVED** [review]

**Pull request merged**